File: javascript/series/series.js
export class Series {

  constructor(digits) {

    this._digits = [...digits].map(Number);
    this._num_digits = this._digits.length;

  }

  get digits() { return this._digits; }

  slices(n) {

    if (n > this._num_digits)
      throw new Error('Slice size is too big.');

    let slices_found = [];
    for (let i = 0; i + n <= this._num_digits; i++)
      slices_found.push(this._digits.slice(i, i + n));
    
    return slices_found;
  }
}

File: java/series/src/main/java/Series.java
import java.util.LinkedList;
import java.util.List;

class Series {

    private final String number;

    Series(String number) {
        this.number = number;
    }

    List<String> slices(int sliceSize) {

        int len = number.length();

        if (sliceSize > len)
            throw new IllegalArgumentException("Slice size is too big.");
        if (sliceSize < 1)
            throw new IllegalArgumentException("Slice size is too small.");

        List<String> slicesList = new LinkedList<>();
        for (int i = 0; i < len - sliceSize + 1; i++) {
            slicesList.add(this.number.substring(i, i + sliceSize));
        }
        return slicesList;
    }
}